src/main/java/de/hhu/propra/sharingplatform/service/OfferService.java

```diff
@@ -4,45 +4,72 @@
 import de.hhu.propra.sharingplatform.model.Item;
 import de.hhu.propra.sharingplatform.model.Offer;
 import de.hhu.propra.sharingplatform.model.User;
-import java.util.Date;
-import lombok.Data;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-@Data
+import java.util.Date;
+
 @Service
 public class OfferService {
 
-    final OfferRepo offerRepo;
+    private OfferRepo offerRepo;
+
+
+    private ContractService contractService;
+
+    private ApiService apiService;
+
+    private PaymentService paymentService;
 
-    final ContractService contractService;
 
     @Autowired
-    public OfferService(ContractService contractService, OfferRepo offerRepo) {
+    public OfferService(ContractService contractService, OfferRepo offerRepo,
+                        ApiService apiService, PaymentService paymentService) {
         this.contractService = contractService;
         this.offerRepo = offerRepo;
+        this.apiService = apiService;
+        this.paymentService = paymentService;
     }
 
-    public void create(Item item, User requester) {
-        Date start = new Date();
-        Date end = new Date();
+    /* Return values:
+     *  0: all gucci
+     *  1: start date >= end date
+     *  2: item not available at given time
+     *  3: not enough money
+     *  4: borrower account banned
+     */
+    public int validate(Item item, User requester, Date start, Date end) {
```

Comment:
Can you throw an HTTP Exception when validate fails? Controller should not have to check the return value...

Reply:
Saw this to late ... we will correct this. Good Point